py-polars/polars/internals/frame.py

```diff
@@ -6377,6 +6321,11 @@ class GroupBy(Generic[DF]):
 
     """
 
+    _df: PyDataFrame
+    _dataframe_class: type[DF]
+    by: str | list[str]
+    maintain_order: bool
```

Comment:
What's the rationale behind defining these types at the class level? Doesn't the `__init__` method take care of this?